src/ProjectNeon/Assets/Features/Effects/SetFeedUpEffect.cs

```diff
@@ -0,0 +1,47 @@
+﻿
+using System;
+
+/**
+ * Puts the decorated effect on queue and sets up a trigger so it is executed after
+ * the next card is played and card is of some specific type.
+ * 
+ * When some "feedType" card is played, this effect is triggered. If the "feedType" is
+ * different from this card, the effect is wasted.
+ * 
+ */
+public class SetFeedUpEffect : Effect
+{
+    private Effect _effect;
+
+    /**
+     * Feed type that triggers the effect execution
+     */
+    private FeedType _feedType;
+
+    public SetFeedUpEffect(Effect origin, FeedType feedType)
+    {
+        _effect = origin;
+        _feedType = feedType;
+    }
+
+    public SetFeedUpEffect(Effect origin, string attribute) : this(origin, (FeedType)Enum.Parse(typeof(FeedType), attribute))
+    {
+
+    }
+
+    public void Apply(Member source, Target target)
+    {
+
+        Effect effect = new UnqueueAfterExceuteEffect(_effect);
```

Comment:
This Queue pop mechanism seems, plus one-shot applying seems like a duplicated code idiom. 

Rather than repeat this, let's build it into a core object, and have the current Objects who use it delegate to the one-shot effect.

Reply:
@SilasReinagel We already have this logic in `QueueEffect`. I'll use it.